### ISSUE TITLE
No OpenGL SOIL package

### ISSUE THREAD

**Issue by AntonioCS** [msys2/MINGW-packages#3369]
Hey guys,

Is there a possibility to include the lib [SOIL](http://www.lonesock.net/soil.html) into a package?

Thanks!

**Comment by ukhf-jm** [comment]
Why not use something like [stb_image](https://github.com/nothings/stb/blob/master/stb_image.h)? It's a single header which you can just include in your project.  
I'd hazard a guess and say that you're following a tutorial like [open.gl](https://open.gl/)? If so then the updated version at [learnopengl.com](https://learnopengl.com) would be better to follow.

**Comment by AntonioCS** [comment]
@ukhf-jm I am trying to compile this game: https://github.com/ryanfleury/global_game_jam_2018
They use SOIL so I though it might be easy to add to msys2.
The learnopengl.com site has comments from 4 years ago. Is that the most up to date site to learn opengl?

**Comment by MenaceInc** [comment]
@AntonioCS yes, the learnopengl.com site is perfectly fine. It focuses on the basics of modern OpenGL (version 3.3 onwards using the core profile) but has newer resources than open.gl.

Regarding that Global Game Jam game, I'm not sure if it can be compiled using mingw-w64 as it uses EFX-Util.h. I'll have a quick look at it though.

**Comment by AntonioCS** [comment]
@MenaceInc Thanks! I did post about it here: https://github.com/Alexpux/MINGW-packages/issues/3368
Thought it might have been a bug or something.

**Comment by MenaceInc** [comment]
@AntonioCS I've managed to get it compiled and running but I'm not happy about how the project is structured right now. You can find my fork of the game here: https://github.com/MenaceInc/global_game_jam_2018

Due to the lack of the SOIL library being available in MINGW-packages, I've had to include the static lib from the release zip from here: http://www.lonesock.net/soil.html

This means that it only compiles and works with the 32bit mingw-w64 compiler. You'll need to run `pacman -Sy mingw-w64-i686-gcc mingw-w64-i686-make mingw-w64-i686-glew mingw-w64-i686-glfw mingw-w64-i686-openal mingw-w64-i686-cmake` from MSYS2 and then build it from the MinGW 32bit shell.

Clone the repo and navigate to the root from within a MinGW 32bit shell then run the following:
```
mkdir build && cd build
cmake .. -DCMAKE_MAKE_PROGRAM=mingw32-make -G "MSYS Makefiles"
mingw32-make
```

You should then be able to navigate to the bin directory and run the game from there.
```
cd ../bin && ./global_game_jam_2018
```

Screenshots:
![image](https://user-images.githubusercontent.com/1106054/36180037-b3737908-1116-11e8-8da2-de3059e92805.png)

![image](https://user-images.githubusercontent.com/1106054/36180031-ab6037a6-1116-11e8-94b1-b43e1fad6da4.png)

**Comment by MenaceInc** [comment]
Something else I will say is that I would NOT recommend using this project to learn from. I know it's from a 48 hour game jam but it is full of terrible, terrible code. It includes .cpp files in the main.cpp unit. I'm pretty sure I saw a few uses of malloc as well as an overuse of preprocessor defines. It is broken in a lot of ways and is not representative of good C++ code. Saying that, I hope you find it useful in some way.

Moving forward, I'd recommend that we do a PR for both SOIL and stb_image.

**Comment by AntonioCS** [comment]
@MenaceInc Wow! Thanks so much for your help.
Yes I agree that this might not be something good to learn from as this is c with classes and not c++ but it does work so I might learn something about game programming and libs that might help me physics or other stuff.

Thanks again for this!

**Comment by MehdiChinoune** [comment]
fixed by #3405
